java-checks/src/main/java/org/sonar/java/checks/NestedSwitchStatementCheck.java

```diff
@@ -24,33 +24,29 @@
 import org.sonar.check.Rule;
 import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
 import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
-import org.sonar.plugins.java.api.tree.CaseGroupTree;
-import org.sonar.plugins.java.api.tree.ClassTree;
 import org.sonar.plugins.java.api.tree.LambdaExpressionTree;
+import org.sonar.plugins.java.api.tree.NewClassTree;
 import org.sonar.plugins.java.api.tree.SwitchStatementTree;
 import org.sonar.plugins.java.api.tree.Tree;
-import org.sonar.plugins.java.api.tree.Tree.Kind;
 
 @Rule(key = "S1821")
 public class NestedSwitchStatementCheck extends IssuableSubscriptionVisitor {
 
   @Override
-  public List<Kind> nodesToVisit() {
+  public List<Tree.Kind> nodesToVisit() {
     return ImmutableList.of(Tree.Kind.SWITCH_STATEMENT);
   }
 
   @Override
   public void visitNode(Tree tree) {
-    SwitchStatementTree switchStatementTree = (SwitchStatementTree) tree;
-    List<CaseGroupTree> cases = switchStatementTree.cases();
-    cases.stream().forEach(c -> c.accept(new NestedSwitchVisitor()));
+    ((SwitchStatementTree) tree).cases().stream().forEach(c -> c.accept(new NestedSwitchVisitor()));
```

Comment:
No need to recreate all the time a new instance of the visitor. The same visitor can be reused for all the case group:

```java
    NestedSwitchVisitor visitor = new NestedSwitchVisitor();
    ((SwitchStatementTree) tree).cases().stream().forEach(c -> c.accept(visitor));
```